qutip/core/gates.py

```diff
@@ -167,19 +186,25 @@ def rx(phi, *, dtype="dense"):
         Quantum object for operator describing the rotation.
 
     """
+    dtype = dtype or settings.core["default_dtype"] or _data.Dense
     return Qobj(
         [
             [np.cos(phi / 2), -1j * np.sin(phi / 2)],
             [-1j * np.sin(phi / 2), np.cos(phi / 2)],
-        ]
+        ],
+        isherm=(phi % (2 * np.pi) <= settings.core["atol"]),
```

Comment:
This is nice but I fear that it may again cause problems with jit. I'm fine just leave it None here. The same for other rotations.